### ISSUE TITLE
Zoom is applied delayed when using capture_continuous

### ISSUE THREAD

**Issue by piveloper** [waveform80/picamera#421]
Hi,
I want to process images with open cv to track an object in front of the camera. To do this more efficiently I resized the image to a lower resolution. When the object gets far away of the camera it is lost in the pixels of the resized image.  To resolve this I want to adjust the zoom of the camera continuously.

Unfortunately when I set the camera.zoom as in following code, I need to run over 3-4 frames in this loop to wait for the zoom becoming active. Is there a way to reduce this delay and get the zoomed frame instantly in the next iteration of the for loop?

`camera.resolution = (1640, 922)
camera.framerate = 25
time.sleep(0.5)
size_resize = (213, 120)
rawCapture = PiRGBArray(camera, size=size_resize)

for frame in camera.capture_continuous(rawCapture, format="bgr", use_video_port=True,
                                               resize=size_resize,
                                               splitter_port=3)
    frame = frame.array
    new_zoom_parameters = get_zoom_parameters_to_center_object(frame)
    camera.zoom = new_zoom_parameters
    rawCapture.truncate(0)`

**Comment by 6by9** [comment]
No, that's not available from the firmware, although may be possible to expose.
Due to the mobile phone origins of the whole camera stack, the requirement was for zoom to be nice and smooth rather than jumping in huge steps.

Even so you haven't a simple mechanism to synchronise requesting frames and updating settings, so you may need a number of frames, particularly if you're using ```use_video_port=True``` which will keep the pipeline running independent of your application consuming the resulting frames.
